src/storage/src/hummock/store/version.rs

```diff
@@ -296,62 +286,33 @@ impl HummockReadVersion {
                     //    are always the suffix of self.staging.imm
 
                     // Check 1)
-                    debug_assert!(self
+                    // Calculate intersection
+                    let staging_imm_ids_from_imms: HashSet<u64> = self
                         .staging
-                        .imm
+                        .data
                         .iter()
-                        .rev()
-                        .is_sorted_by_key(|imm| imm.batch_id()));
```

Comment:
I think the imm_ids in `self.staging.data` are still sorted. Why do we remove this assertion?

Reply:
Because some of them may be sstable file, and sstable file does not have a batch id.